DigitalLearningSolutions.Web.Tests/Controllers/Register/ClaimAccountControllerTests.cs

```diff
@@ -399,204 +419,163 @@ public void
+            string email,
+            string code
+        )
         {
             // When
-            var result = controllerWithLoggedInUser.LinkDlsAccount();
+            var result = controllerWithLoggedInUser.LinkDlsAccountPost(email, code);
 
             // Then
             result.Should().BeRedirectToActionResult().WithControllerName("LearningSolutions")
                 .WithActionName("AccessDenied");
         }
 
         [Test]
-        public void
-            LinkDlsAccountPost_when_the_logged_in_user_already_has_a_delegate_account_at_the_centre_redirects_to_AccountAlreadyExists()
+        public void LinkDlsAccountPost_with_no_existing_user_redirects_to_AccessDenied()
         {
             // Given
-            var expectedModel = GetViewModel();
-            var userEntity = new UserEntity(
-                UserTestHelper.GetDefaultUserAccount(),
-                new List<AdminAccount>(),
-                new List<DelegateAccount> { UserTestHelper.GetDefaultDelegateAccount(centreId: DefaultCentreId) }
-            );
-
-            controllerWithLoggedInUser.TempData.Set(expectedModel);
-
             A.CallTo(
-                () => userService.GetUserById(DefaultLoggedInUserId)
-            ).Returns(userEntity);
+                () => userDataService.GetUserIdAndCentreForCentreEmailRegistrationConfirmationHashPair(
+                    DefaultEmail,
+                    DefaultCode
+                )
+            ).Returns((null, null, null));
 
             // When
-            var result = controllerWithLoggedInUser.LinkDlsAccount();
+            var result = controllerWithLoggedInUser.LinkDlsAccountPost(DefaultEmail, DefaultCode);
 
             // Then
-            result.Should().BeRedirectToActionResult()
-                .WithActionName("AccountAlreadyExists");
+            A.CallTo(
+                () => userDataService.GetUserIdAndCentreForCentreEmailRegistrationConfirmationHashPair(
+                    DefaultEmail,
+                    DefaultCode
+                )
+            ).MustHaveHappenedOnceExactly();
 
-            controllerWithLoggedInUser.TempData.Peek<ClaimAccountViewModel>()!.Should().BeEquivalentTo(expectedModel);
+            result.Should().BeRedirectToActionResult().WithControllerName("LearningSolutions")
+                .WithActionName("AccessDenied");
         }
 
         [Test]
         public void
-            LinkDlsAccountPost_when_the_claim_email_address_matches_the_primary_email_of_another_user_redirects_to_WrongUser()
+            LinkDlsAccountPost_when_the_logged_in_user_already_has_a_delegate_account_at_the_centre_redirects_to_AccountAlreadyExists()
         {
             // Given
-            var expectedModel = GetViewModel(emailIsTaken: true);
-
-            controllerWithLoggedInUser.TempData.Set(expectedModel);
+            GivenValidViewModel(loggedInUserId: DefaultLoggedInUserId);
+            GivenUserEntity();

```

Comment:
I don't understand how these methods equate to the logged in user having a delegate account at the same centre as the account to be claimed.
Can you make the names / parameters a bit more explanatory?

Reply:
I have changed
- `GivenValidViewModel` -> `GivenClaimAccountViewModel` 
- `GivenUserEntity` -> `GivenLoggedInUserWithDelegateAccount(int userId, int centreIdForDelegateAccount)`
- The calls to these functions in the `when_the_logged_in_user_already_has_a_delegate_account_at_the_centre` test to explicitly supply user id (for logged in user) and centre id